taichi/transforms/constant_fold.cpp

```diff
@@ -177,11 +177,32 @@ class ConstantFold : public BasicStmtVisitor {
       auto evaluated =
           Stmt::make<ConstStmt>(LaneAttribute<TypedConstant>(new_constant));
       stmt->replace_with(evaluated.get());
-      modifier.insert_before(stmt, VecStatement(std::move(evaluated)));
+      modifier.insert_before(stmt, std::move(evaluated));
       modifier.erase(stmt);
     }
   }
 
+  void visit(BitExtractStmt *stmt) override {
+    auto input = stmt->input->cast<ConstStmt>();
+    if (!input)
+      return;
```

Comment:
```suggestion
    if (!input || input->element_type() == DataType::i32)
      return;
```

Reply:
Why not do constant folding when the `input` is `i32`?